fix: add OverrideDeviceInstanceConfig logic in dmiserver MapperRegister

HaojieZhang6848 opened the pull request:
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

In release 1.22, the kubeedge supports Device CRD to share DeviceModel CRD `protocolConfigData` and `visitors` fields #6457 #6458 [link](https://github.com/kubeedge/kubeedge/blob/master/CHANGELOG/CHANGELOG-1.22.md#device-model-update-based-on-thing-model-and-product-concept). This logic is implemented in `DMIWorker.overrideDeviceInstanceConfig`. This function is called when the cloudcore send Device CRD creation or update message to the edgecore. The edgecore use it to merge `protocolConfigData` and `visitors` from DeviceModel CRD to Device CRD, send it to mapper, and save merged Device CRD into dmiCache. However, the `MapperRegister` implementation in DMIServer does not have this logic. Hence, when the `MapperRegister` is called after the restart of edgecore, the edgecore will send the unmerged Device CRD to mapper. This is problematic.

In this PR, I add the logic of `OverrideDeviceInstanceConfig` in `MapperRegister` implementation to fix the problem.

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note

```

Shelley-BaoYue commented on edge/pkg/devicetwin/dtmanager/dmiworker_test.go:
```diff
@@ -52,6 +52,11 @@ func TestDMIWorker_overrideDeviceInstanceConfig(t *testing.T) {
 							"timeout":    5000,
 							"retryTimes": 3,
 							"baudRate":   9600,
+							"items": []interface{}{"item1", "item2"},
```
Failed integration test is not related to your PR, you don't need to solve it here.   
DCO failed, you need to sign off your commit.

Shelley-BaoYue commented:
There are still some problem with `go lint`, please see more details in CI.

HaojieZhang6848 commented:
> There are still some problem with `go lint`, please see more details in CI.

Fixed in commit [bc3f5cf](https://github.com/kubeedge/kubeedge/commit/bc3f5cf9c2b5268892d9a357cc0e2d6d30b85ed3). Now, `make e2e`, `make test PROFILE=y` and `make lint` can all pass on my local environment.
In addition, I moved unit tests related to OverrideDeviceInstanceConfig to dmiserver package (where the OverrideDeviceInstanceConfig currently in), to ensure the test coverage of `OverrideDeviceInstanceConfig` is correctly reported.

Shelley-BaoYue commented on edge/pkg/devicetwin/dmiserver/server.go:
```diff
@@ -67,6 +67,142 @@ type DMICache struct {
 }
 
+// overrideDeviceInstanceConfig overrides device instance configuration with model defaults
+func (dmiCache *DMICache) OverrideDeviceInstanceConfig(device *v1beta1.Device) error {
```
Since we need cherry-pick this PR to release-1.22, how about put this optimization work into a seperate PR? @WillardHu

WillardHu commented:
/lgtm
